### PR TITLE
COE: Add end-to-end tests

### PR DESCRIPTION
**Pull request by Mottie**
Original Ticket [#42549](https://github.com/department-of-veterans-affairs/va.gov-team/issues/42549)

URL of change:  [COE form](http://localhost:3001/housing-assistance/home-loans/request-coe-form-26-1880/)

# Background

Added missing Cypress e2e tests to the Certificate of Eligibility form.

In addition, we needed to remove a `<dl>` wrapper from the service history page, and move it inside the platform `ReadOnlyArrayField` so that the axe check on the service history page would work (use [maximal-data](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/lgy/coe/form/tests/fixtures/data/maximal-test.json) to test) on `http://localhost:3001/housing-assistance/home-loans/request-coe-form-26-1880/service-history` page

## Steps to test

- Switch to the test this branch
- Open Cypress `yarn cy:open`
- Find COE form test (not COE status)
- Run maximal & minimal tests in panel

## Code changes

- Add Cypress e2e tests using maximal & minimal data
- Add mock data

## How was your code tested

Cypress test panel. And pair programmed!

## Acceptance criteria
- [x] Added Cypress e2e tests
- [x] All tests passing

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/test/js/review/ReadOnlyArrayField.unit.spec.jsx [hunk 1]
```diff
@@ -48,5 +48,6 @@ describe('Schemaform review: ReadOnlyArrayField', () => {
     expect(tree.subTree('.schemaform-array-readonly-header').text()).to.contain(
       'Item',
     );
+    expect(tree.everySubTree('.review').length).to.equal(1);
```
Comment (Mottie) [inline src/platform/forms-system/test/js/review/ReadOnlyArrayField.unit.spec.jsx:51]: This is using SkinDeep testing, so I didn't check the `dl > div > dd` or `dl > div > dt` structure specifically

File: src/platform/forms-system/src/js/review/ReadOnlyArrayField.jsx [hunk 2]
```diff
@@ -21,31 +21,37 @@ class ReadOnlyArrayField extends React.Component {
...
+            >
+              <div className="row small-collapse">
+                <div className="small-12 columns">
+                  <h5 className="schemaform-array-readonly-header">
+                    {uiOptions.itemName}
+                  </h5>
+                  {/* outer wrapper needs uiOptions.customTitle = ' ' to prevent
+                    * rendering of a <dl> around the schemaform-field-container
+                    */}
+                  <dl className="review">
```
Comment (Mottie) [inline src/platform/forms-system/src/js/review/ReadOnlyArrayField.jsx:54]: I didn't research if this read-only component is used elsewhere...

File: src/applications/lgy/coe/shared/actions/index.js [hunk 3]
```diff
@@ -11,6 +12,7 @@ export const GET_COE_URL_FAILED = 'GET_COE_URL_FAILED';
 const COE_DOWNLOAD_URI = '/coe/download_coe';
 const COE_STATUS_URI = '/coe/status';
 
+//
```
Comment (scottymeyers) [inline src/applications/lgy/coe/shared/actions/index.js:15]: didn't finish your thought here

**Review by scottymeyers — APPROVED** [review]

**Review by bjmccotter7192 — COMMENTED** [review]
As far as I can tell this looks good 👍

**Review by harshilmoradia — APPROVED** [review]

**Pull request merged**